packages/@aws-cdk/cdk/lib/core/tokens.ts

```diff
@@ -29,13 +48,59 @@ export class Token {
 
         return value;
     }
+
+    /**
+     * Return a reversible string representation of this token
+     *
+     * If the Token is initialized with a literal, the stringified value of the
+     * literal is returned. Otherwise, a special quoted string representation
+     * of the Token is returned that can be embedded into other strings.
+     *
+     * Strings with quoted Tokens in them can be restored back into
+     * complex values with the Tokens restored by calling `resolve()`
+     * on the string.
+     */
+    public toString(): string {
+        const valueType = typeof this.valueOrFunction;
+        // Optimization: if we can immediately resolve this, don't bother
+        // registering a Token.
+        if (valueType === 'string' || valueType === 'number' || valueType === 'boolean') {
+            return this.valueOrFunction.toString();
+        }
+
+        if (this.tokenKey === undefined) {
+            this.tokenKey = TOKEN_STRING_MAP.register(this, this.displayName);
+        }
+        return this.tokenKey;
+    }
+
+    /**
+     * Turn this Token into JSON
+     *
+     * This gets called by JSON.stringify(). We want to prohibit this, because
+     * it's not possible to do this properly, so we just throw an error here.
+     */
+    public toJSON(): any {
```

Comment:
There’s a leak in the docs... talks about CfN?

Reply:
Yes. I was unsure of how to phrase this error message to be helpful without mentioning the specifics, so I erred on the side of helpful.